scripts/generate_proto_stubs.sh

```diff
@@ -73,7 +73,8 @@ PROTO_FILES=$(grep "GenProto.*google" $PYTHON_PROTOBUF_DIR/python/setup.py | \
 # shellcheck disable=SC2086
 protoc_install/bin/protoc --proto_path="$PYTHON_PROTOBUF_DIR/src" --mypy_out="relax_strict_optional_primitives:$REPO_ROOT/stubs/protobuf" $PROTO_FILES
 
-ruff check "$REPO_ROOT/stubs/protobuf" --fix-only
+# --unsafe-fixes to apply outdated-version-block (UP036)
+ruff check "$REPO_ROOT/stubs/protobuf" --fix-only --unsafe-fixes
```

Comment:
I feel slightly queasy about enabling `--unsafe-fixes` here, as we have a lot of rules enabled in our `pyproject.toml`, and I haven't audited all the ones that have unsafe fixes to see just how unsafe they are. Also:
- We might add more rules to `pyproject.toml` in the future; we'll probably forget to audit those rules for unsafe fixes, given that we don't have `--unsafe-fixes` enabled in our CI elsewhere
- It feels a little weird to have a flag in this script that we don't use for ruff in CI; that might be unexpected for people running this script
- With this script, the autofix isn't even shown as a separate commit on the resulting PR; it's just "invisibly" applied before the PR is even made. So I worry that people might not even notice that ruff made changes

Reply:
Would it be fine if I added this line instead?
`ruff check "$REPO_ROOT/stubs/protobuf" --select=UP036 --fix-only --unsafe-fixes`
With that I wouldn't even need config change in this PR

Reply:
That SGTM!